portals/publisher/src/main/webapp/source/src/app/components/Apis/Create/CreateAPIWithAI/APICreateWithAI.jsx

```diff
@@ -0,0 +1,291 @@
+ * under the License.
+ */
+import React, { useState, useRef, useEffect } from 'react';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import { styled } from '@mui/material/styles';
+import ApiChatPoweredBy from './components/ApiChatPoweredBy';
+import ApiChatBanner from './components/ApiChatBanner';
+import ApiChatExecute from './components/ApiChatExecute';
+import ApiChatResponse from './components/ApiChatResponse';
+import SampleQueryCard from './components/SampleQueryCard';
+import LinearDeterminate from './components/LinearDeterminate';
+import DisplayCode from './components/DisplayCode';
+import WelcomeMessage from './components/WelcomeMessage';
+import LoadingDots from './components/LoadingDots';
+import API from 'AppData/api';
+
+/**
+ * Renders the Create API with AI UI.
+ * @returns {JSX} Create API with AI page to render.
+ */
+const ApiCreateWithAI = () => {
+    const [sessionId, setSessionId] = useState(null);
+    const [inputQuery, setInputQuery] = useState('');
+    const [lastQuery, setLastQuery] = useState('');
+    const [finalOutcome, setFinalOutcome] = useState('');
+    const [apiType, setApiType] = useState('');
+    const [finalOutcomeCode, setFinalOutcomeCode] = useState('');
+    const [executionResults, setExecutionResults] = useState([]);
+    const [messages, setMessages] = useState([]);
+    const [taskId, setTaskId] = useState('');
+    const [taskStatus, setTaskStatus] = useState('');
+    const [loading, setLoading] = useState(false); 
+    const [isSuggestion, setIsSuggestion] = useState('');
+    const [selectedTitles, setSelectedTitles] = useState([]);
+    const [paths, setPaths] = useState([]);
+    const [apiTypeSuggestion, setApiTypeSuggestion] = useState('');
+    const [missingValues, setMissingValues] = useState('');
+
+    const chatContainerRef = useRef(null);
+
+    useEffect(() => {
+        if (chatContainerRef.current) {
+            const element = chatContainerRef.current;
+            element.scrollTop = element.scrollHeight;
+        }
+    }, [messages]);
+    
+    const handleQueryChange = (event) => {
+        const { value } = event.target;
+        setInputQuery(value);
+    };
+
+    const handleExecute = async () => {
+        if (inputQuery.length !== 0) {
+            const query = inputQuery;
+            setInputQuery('');
+            setLastQuery(inputQuery);    
+            setTaskStatus('IN_PROGRESS');
+
+            setMessages((prevMessages) => [
+                ...prevMessages,
+                { role: 'user', content: query },
+            ]);
+
+            if (!sessionId) {
+                const newSessionId = generateSessionId();
+                setSessionId(newSessionId);
+                sendInitialRequest(query, newSessionId);
+            } else {
+                sendInitialRequest(query, sessionId);
+            }
+        }
+    };
+
+    const handleExecuteSampleQuery = async (query) => {
```

Comment:
The `handleExecute` and `handleExecuteSampleQuery` functions look same in most of the parts. Can have one method and do the logic by checking whether the `query` param is `undefined` or not.